src/coreclr/jit/morph.cpp

```diff
@@ -7453,7 +7457,7 @@ GenTree* Compiler::fgMorphTailCallViaHelpers(GenTreeCall* call, CORINFO_TAILCALL
         JITDUMP("Removing retbuf");
 
         call->gtArgs.Remove(call->gtArgs.GetRetBufferArg());
-        call->gtCallMoreFlags &= ~GTF_CALL_M_RETBUFFARG;
+        call->gtCallMoreFlags &= ~(GTF_CALL_M_RETBUFFARG | GTF_CALL_M_RETBUFFARG_LCLOPT);
```

Comment:
In general the rule is that if you remove `GTF_CALL_M_RETBUFFARG_LCLOPT`, you need to address-expose the underlying local (since it makes it invisible to liveness), I suppose here we can elide that because the logical definition occurs outside the method (since it is a tailcall)?

Perhaps worth a comment (or not - maybe it is self-evident).

Reply:
Right, things have already gone very wrong if think we have a tailcall with a retbuf that can be optimized with this optimization.
I will just revert this change, we will hit the assert inside of `gtCallGetDefinedRetBufLclAddr` if the flag is set and we don't have a retbuf.